### PR TITLE
squad indices / handwritten teacher update

### PR DESCRIPTION
**Pull request by alexholdenmiller**
Add support for batching and metrics to squad handwritten teacher, and add answer indices

### CODE REVIEW

File: parlai/tasks/squad/agents.py [hunk 1]
```diff
@@ -15,62 +15,116 @@
...
             self.epochDone = True
 
-        return {
+        action = {
+            'id': 'squad',
             'text': context + '\n' + question,
-            'labels': answers,
-            'episode_done': True
+            'episode_done': True,
+            'answer_starts': answer_starts
```
Comment (jaseweston) [inline parlai/tasks/squad/agents.py]: hmm..slightly weird name but i guess that's what they called it... why plural though?

Comment (alexholdenmiller) [reply parlai/tasks/squad/agents.py]: one start for each label :) (there are several labels for every question)

**Comment by alexholdenmiller** [comment]
some of this functionality should maybe be moved to a "SupervisedTeacher" or "IndexedTeacher" which could be a parent of DialogTeacher--then people could inherit the batch indexing

**Comment by alexholdenmiller** [comment]
at least now this class is a more detailed example

**Pull request merged**